jdbc-company/src/dao/CompanyDaoImpl.java

```diff
@@ -0,0 +1,101 @@
+package dao;
+import com.bobocode.exception.DaoOperationException;
+import core.Company;
+
+import java.sql.*;
+import javax.sql.DataSource;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by hamster on 21.08.2017.
+ */
+public class CompanyDaoImpl implements CompanyDao {
+    private DataSource dataSource;
+
+    public CompanyDaoImpl(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    @Override
+    public void save(Company company) {
+        try {
+            Connection connection = dataSource.getConnection();
+            Statement statement = connection.createStatement();
+            PreparedStatement createCompanyString = prepareInsertStatement(connection, company);
+            statement.executeUpdate(createCompanyString.toString()); //вот тут непонятно что собственно делает в коде Тараса executeInsert
```

Comment:
It wraps `executeUpdate() `call with `if` clause to check whether a row was inserted. It was created to show you that you can check the **number of rows that were affected by your update**.

Reply:
Yeah, probably. In general "HOW" is always up to you. You have code examples. However, you need to understand, that requirements will never be clear. Software engineering is about finding solutions. It's not about typing code using given instruction. I'll try to add more comments, but...

> -do we need use existing classes? which of them? why?
> -do we need create new module for our homework or use existing? why?
> -how we can create pull request?

These are questions you should ask yourself, and try to find the answer. I will stand behind you ;)